### ISSUE TITLE
Crash app when set UserAttribute in Application

### ISSUE THREAD

**Issue by dung90it** [Instabug/Instabug-Android#202]
### crashes on app start
Caused by: android.database.sqlite.SQLiteConstraintException: 
  at android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId (Native Method)
  at android.database.sqlite.SQLiteConnection.executeForLastInsertedRowId (SQLiteConnection.java:995)
  at android.database.sqlite.SQLiteSession.executeForLastInsertedRowId (SQLiteSession.java:788)
  at android.database.sqlite.SQLiteStatement.executeInsert (SQLiteStatement.java:86)
  at android.database.sqlite.SQLiteDatabase.insertWithOnConflict (SQLiteDatabase.java:2021)
  at com.instabug.library.internal.storage.cache.db.SQLiteDatabaseWrapper.insertWithOnConflict (SQLiteDatabaseWrapper.java:62)
  at com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributesDbHelper.insert (UserAttributesDbHelper.java:47)
  at com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributeCacheManager.insert (UserAttributeCacheManager.java:17)
  at com.instabug.library.Instabug.setUserAttribute (Instabug.java:587)
  at com.fuelcloud.skynet.SkynetApp.onCreate (SkynetApp.kt:57)
### Instabug integration code
 Instabug.Builder(this@SkynetApp, WebServiceApi.INSTABUG_KEY)
                    .setInvocationEvents(InstabugInvocationEvent.TWO_FINGER_SWIPE_LEFT)
                    .setReproStepsState(State.ENABLED)
                    .build()
            Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED)
            FeatureRequests.setEmailFieldRequired(false, ActionType.ADD_COMMENT_TO_FEATURE)
            BugReporting.setOptions(Option.EMAIL_FIELD_OPTIONAL)
            BugReporting.setReportTypes(BugReporting.ReportType.BUG)
            Instabug.setSessionProfilerState(Feature.State.ENABLED)
            Chats.setState(Feature.State.DISABLED)
            Instabug.setLocale(Locale.US)
            Instabug.setUserAttribute("App Name", "FUELCLOUD")

**Comment by mzelzoghbi** [comment]
Hey @dung90it, Thanks for reaching out.
We tried the above steps on different devices and it worked fine, Is this happening on this device only **(Galaxy S10+)** or many devices? If it is happening on all devices it will be great to share with us a sample project with the issue.

**Comment by dung90it** [comment]
@mzelzoghbi some devices we have this issues
Brand: samsung
Model: Galaxy S5
Version: 6.0.1

Brand: samsung
Model: Galaxy S7
Version: 8.0.0

Brand: samsung
Model: Galaxy A5(2017)
Version: 8.0.0

Brand: samsung
Model: Galaxy J7 Crown
Version: 8.0.0

Brand: samsung
Model: Galaxy J3 Eclipse
Version: 8.1.0

Brand: samsung
Model: Galaxy J7 Perx
Version: 8.1.0

**Comment by TankJK** [comment]
Hi,

I am also facing same issue. Lots of users are having this crash.

Caused by android.database.sqlite.SQLiteConstraintException: FOREIGN KEY constraint failed (code 787)
#################################################################
Error Code : 787 (SQLITE_CONSTRAINT_FOEIGNKEY)
Caused By : Abort due to constraint violation.
	(FOREIGN KEY constraint failed (code 787))
#################################################################
       at android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId(SQLiteConnection.java)
       at android.database.sqlite.SQLiteConnection.executeForLastInsertedRowId + 953(SQLiteConnection.java:953)
       at android.database.sqlite.SQLiteSession.executeForLastInsertedRowId + 788(SQLiteSession.java:788)
       at android.database.sqlite.SQLiteStatement.executeInsert + 86(SQLiteStatement.java:86)
       at android.database.sqlite.SQLiteDatabase.insertWithOnConflict + 1904(SQLiteDatabase.java:1904)
       at com.instabug.library.internal.storage.cache.db.SQLiteDatabaseWrapper.insertWithOnConflict + 62(SQLiteDatabaseWrapper.java:62)
       at com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributesDbHelper.insert + 47(UserAttributesDbHelper.java:47)
       at com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributeCacheManager.insert + 17(UserAttributeCacheManager.java:17)


Please do the needful at earliest.

Thanking you in advance,
Jignesh

**Comment by mzelzoghbi** [comment]
Thanks, @dung90it @TankJK  for reporting this, we will take a look and get back to you ASAP.

**Comment by amrbrkt** [comment]
@dung90it @TankJK would you please let us know if any if your apps are running on multiple processes?

**Comment by jdamcd** [comment]
We're also seeing this crash for large numbers of users on upgrading the Instabug SDK from 8.4.0 to 8.5.0. Our app does not use multiple processes.

**Comment by dung90it** [comment]
@amrbrkt my app uses multiple processes, this crash occurs when app upgrading the Instabug SDK from 8.4.0 to 8.5.0.

**Comment by amrbrkt** [comment]
@jdamcd @TankJK Please give `8.5.0.2181-SNAPSHOT` and let us know if you still have similar issues. 

**To use the snapshot:**
Add these to your app-level `build.gradle` file

```
dependencies {
  compile 'com.instabug.library:instabug:4.x.y.z-SNAPSHOT'
}
```
```
repositories {
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}

```

**Comment by amrbrkt** [comment]
This fix has been released, please update to the recent version `8.6.3`. Please don't hesitate to get back to us if the problem still exists.

**Comment by andy3466** [comment]
Hi, I am still facing same issue after updating to the recent version 8.6.3. Lots of users are having this crash.


Caused by android.database.sqlite.SQLiteConstraintExceptionFOREIGN KEY constraint failed (code 787 SQLITE_CONSTRAINT_FOREIGNKEY[787])

android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId (SQLiteConnection.java)
android.database.sqlite.SQLiteDatabase.insertWithOnConflict + 2021 (SQLiteDatabase.java:2021)
com.instabug.library.internal.storage.cache.db.SQLiteDatabaseWrapper.insertWithOnConflict + 62 (SQLiteDatabaseWrapper.java:62)
com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributesDbHelper.insert + 46 (UserAttributesDbHelper.java:46)
 com.instabug.library.internal.storage.cache.db.userAttribute.UserAttributeCacheManager.insert + 19 (UserAttributeCacheManager.java:19)
com.instabug.library.util.a.a$5.a + 71 (a.java:71)
com.instabug.library.util.a.a$5.a + 66 (a.java:66)
com.instabug.library.util.a.c.a + 54 (c.java:54)
com.instabug.library.Instabug.setUserAttribute + 538 (Instabug.java:538)
live.free.tv.GlobalApplication.onCreate + 61 (GlobalApplication.java:61)
android.app.Instrumentation.callApplicationOnCreate + 1154 (Instrumentation.java:1154)
com.android.internal.os.ZygoteInit.main + 965 (ZygoteInit.java:965)

Instabug integration code
Instabug.Builder(this, INSTABUG_KEY)
.setInvocationEvents(InstabugInvocationEvent.NONE)
.build()

Devices which have this issue : 

<img width="916" alt="螢幕快照 2019-09-23 下午3 08 58" src="https://user-images.githubusercontent.com/16247871/65407179-49c6fa80-de14-11e9-9243-25a126113edd.png">

**Comment by mzelzoghbi** [comment]
Hi @andy3466,
We noticed that the fix we shipped in 8.6.3 didn't tackle all the cases that caused this crash, so we shipped another solid fix in **8.7.0**. 

Please accept our apologies for the inconveniences we may have caused. 

ThankYou!
